docs/source/notes/cuda.rst

```diff
@@ -75,18 +75,32 @@ follows::
       x = x.cuda()
       net.cuda()
 
-To avoid the if statement for tensors, an alternative is to have a default
+When working with multiple GPUs on a system, you can use the
+`CUDA_VISIBLE_DEVICES` environment flag to manage which GPUs are available to
+PyTorch. To manually control which GPU a tensor is created on, the best practice
+is to use the `torch.cuda.device()` context manager::
+
+    torch.cuda.set_device(0)
+    print("Outside device is 0")  # On device 0
+    with torch.cuda.device(1):
+        print("Inside device is 1")  # On device 1
+    print("Outside device is still 0")  # On device 0
+
+When creating tensors, an alternative to the if statement is to have a default
 datatype defined, and cast all tensors using that. An example when using a
 dataloader would be as follows::
 
     dtype = torch.cuda.FloatTensor
     for i, x in enumerate(train_loader):
         x = Variable(x.type(dtype))
 
-If you have a tensor and would like to create a new tensor of the same type,
-then you can use the `.new()` function, which acts the same as a normal tensor
-constructor. This is the recommended practice when creating modules in which
-new tensors/variables need to be created internally during the forward pass::
+If you have a tensor and would like to create a new tensor of the same type on
```

Comment:
I think a better way to structure this would be to list all methods, and then mention that they all have the downside that they depend on the GPU context. Then, show `torch.cuda.device` (and please remove `torch.cuda.set_device`, it's mostly meant to be a helper for IPython notebooks and not for scripts in general where you can use `CUDA_VISIBLE_DEVICES`). Then, mention `new` and say that it preserves everything

Reply:
Done. I now also noted that `ones_like` and `zeros_like` also preserve device - can you confirm?

Reply:
Yes, that's true